.github/workflows/pull-fork.yml

```diff
@@ -0,0 +1,89 @@
+name: Pull request build
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+defaults:
+  run:
+    working-directory: src/client
+
+env:
+  BUCKET_WEB: reactive-trader-web-builds
+  BUCKET_OPENFIN: reactive-trader-openfin-builds
+  BUCKET_FINSEMBLE: reactive-trader-finsemble-builds
+
+jobs:
+  build:
+    name: Build & deploy
+
+    runs-on: ubuntu-20.04
+
+    if: "github.event.pull_request.header.repo.fork"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set Node to v16
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Set up GCloud
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          project_id: ${{ secrets.GOOGLE_PROJECT_ID }}
+          service_account_key: ${{ secrets.GOOGLE_PRIVATE_KEY }}
+
+      - name: Install
+        run: npm ci
+
+      - name: Verify Quality (type checking, linting, format checking, tests)
+        run: |
+          npm run verify
+
+      - name: Build (web)
+        env:
+          DOMAIN: https://web.env.reactivetrader.com
+          URL_PATH: /pull/${{ github.event.number }}
+          VITE_BUILD_VERSION: ${{ github.sha }}
+          ENVIRONMENT: env
+        run: |
+          npm run build
+          npm run storybook:build
+
+      - name: Build (OpenFin)
+        env:
+          DOMAIN: https://openfin.env.reactivetrader.com
+          URL_PATH: /pull/${{ github.event.number }}
+          VITE_BUILD_VERSION: ${{ github.sha }}
+          ENVIRONMENT: env
+        run: npm run openfin:build
+
+      - name: Build (OpenFin - Workspace)
+        env:
+          DOMAIN: https://openfin.env.reactivetrader.com
+          URL_PATH: /pull/${{ github.event.number }}/workspace
+          VITE_BUILD_VERSION: ${{ github.sha }}
+          ENVIRONMENT: env
+        run: |
+          npm ci
+          npm run build
+        working-directory: src/workspace
+
+      - name: Build (Finsemble)
```

Comment:
side note: consider removing this from everyday workflows .. we rarely need this, but we should address it in upcoming [Finsemble upgrade ticket](https://adaptive.kanbanize.com/ctrl_board/18/cards/4823/details/)